contracts/utils/Create2.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Create2
+ *
+ * @dev Utility library that focus on the use of CREATE2 EVM opcode for
+ * contracts deployment. It also provides a function to precompute the address
+ * where the smart contracts with the specified salt and bytecode would be
+ * deployed.
+ */
+library Create2 {
+
+    /**
+     * @dev Deploy contract with CREATE2
+     * @param salt The salt used to the contract address computation
+     * @param code The bytecode of of the contract to be deployed
+     */
+    function deploy(bytes32 salt, bytes memory code) internal returns (address) {
+        address addr = _deploy(salt, code);
+        return addr;
+    }
+
+
+    /**
+     * @dev Function to compute the address of a contract created with CREATE2.
+     * @param salt The salt used to the contract address computation
+     * @param code The bytecode of the contract to be deployed
+     * @return the computed address of the smart contract.
+     */
+    function computeAddress(
+        bytes32 salt, bytes memory code
+    ) internal view returns (address) {
+        return computeAddressWithDeployer(address(this), salt, code);
+    }
+
+    /**
+     * @dev Function to compute the address of a contract created with CREATE2
+     * with the deployer address.
+     * @param deployer the address of the contract that will deploy the contract
+     * @param salt The salt used to the contract address computation
+     * @param code The bytecode of the contract to be deployed
+     * @return the computed address of the smart contract.
+     */
+    function computeAddressWithDeployer(
+        address deployer, bytes32 salt, bytes memory code
```

Comment:
```suggestion
        bytes32 salt, bytes memory code, address deployer
```

Nitpicky, but I think this order makes it more consistent with `computeAddress`.